make.js

```diff
@@ -177,229 +180,255 @@ CLI.gendocs = function() {
 //
 // ex: node make.js build
 // ex: node make.js build --task ShellScript
+// ex: node make.js build --task ShellScript --node Node20
 //
-CLI.build = function() 
+CLI.build = function(/** @type {{ node: string; task: string }} */ argv) 
 {
     if (process.env.TF_BUILD) {
         fail('Please use serverBuild for CI builds for proper validation');
     }
 
     callGenTaskDuringBuild = true;
-    CLI.serverBuild();
+    CLI.serverBuild(argv);
 }
 
-CLI.serverBuild = function() {
+CLI.serverBuild = function(/** @type {{ node: string; task: string }} */ argv) {
     ensureBuildTasksAndRemoveTestPath();
-
-    ensureTool('tsc', '--version', 'Version 4.0.8');
+    ensureNvmInstalled();
+    ensureTool('tsc', '--version', 'Version 4.0.2');
     ensureTool('npm', '--version', function (output) {
         if (semver.lt(output, '5.6.0')) {
             fail('Expected 5.6.0 or higher. To fix, run: npm install -g npm');
         }
     });
 
-    const removeNodeModules = taskList.length > 1;
     const allTasks = getTaskList(taskList);
 
     // Need to validate generated tasks first
     const makeOptions = fileToJson(makeOptionsPath);
 
     util.processGeneratedTasks(baseConfigToolPath, taskList, makeOptions, callGenTaskDuringBuild);
 
-    allTasks.forEach(function(taskName) {
-        let isGeneratedTask = false;
-        banner('Building: ' + taskName);
+    const nodeVersion = argv.node ? argv.node : "Default";
+    const allTasksNode20 = allTasks.filter((taskName) => {
+        return taskName.endsWith("Node20");
+    });
+    const allTasksDefault = allTasks.filter((taskName) => {
+        return !taskName.endsWith("Node20");
+    });
 
-        // If we have the task in generated folder, prefer to build from there and add all generated tasks which starts with task name
-        var taskPath = path.join(genTaskPath, taskName);
-        if (fs.existsSync(taskPath)) {
-            // Need to add all tasks which starts with task name
-            console.log('Found generated task: ' + taskName);
-            isGeneratedTask = true;
-        } else {
-            taskPath = path.join(tasksPath, taskName);
+    if (nodeVersion == "Node20") {
+        ensureTool('node', '--version', `v${node20Version}`);
+        allTasksNode20.forEach(taskName => buildTask(taskName, allTasksNode20.length));
+    } else if (nodeVersion == "Default") {
+        if (allTasksNode20) {
+            console.log(
+            "==========================================\n" + 
+            "IMPORTANT NOTE: There are additional tasks that need to be build with a different node configuration. \n" + 
+            "Unfortunately, we cannot switch node versions while running make.js.  Run the following commands to continue: \n" +
+            `nvm use ${node20Version}\n` +
+            `node make.js build --task ${argv.task} --node Node20\n` + 
```

Comment:
>--task ${argv.task} [](http://example.com/codeflow?start=32&length=19)

[not blocking, may handle in another pr] this doesn't handle when --task isn't specified (all tasks built)